daemon/api_system_volumes.go

```diff
@@ -87,3 +98,29 @@ func postSystemVolumesActionGenerateRecoveryKey(c *Command) Response {
 		"key-id":       keyID,
 	})
 }
+
+var keyIDRequiredErr = BadRequest("system volume action requires key-id to be provided")
+
+func postSystemVolumesActionReplaceRecoveryKey(c *Command, req *systemVolumesActionRequest) Response {
+	if req.KeyID == "" {
+		return keyIDRequiredErr
```

Comment:
don't think the error is going to be used anywhere else, so you may as well skip the package variable 

```suggestion
		return BadRequest("system volume action requires key-id to be provided")
```